### ISSUE TITLE
Call StableDiffusionPipeline object with wrong data type

### ISSUE THREAD

**Issue by chesha1** [huggingface/diffusers#1677]
### Describe the bug

When I call pipe object, it throws Runtime Error: expected scalar type half but found float
I am not sure whether it is the dtype of latent, so I manually define the dtype of latents to fp16 but gets the same error.

### Reproduction

import torch
from diffusers import StableDiffusionPipeline

device = "cuda"
model_id = "./sd-2-1/stable-diffusion-2-1"

pipe = StableDiffusionPipeline.from_pretrained(model_id, revision="fp16", torch_dtype=torch.float16, use_auth_token=True)
pipe = pipe.to(device)

prompt = "anime girl, beautiful, big breast, blue eyes, long hair, masterpiece"
num_images_per_prompt = 1

generator = torch.Generator(device=device)
latents = None
seeds = []
for _ in range(num_images_per_prompt):
    seed = generator.seed()
    seeds.append(seed)
    generator = generator.manual_seed(seed)
    image_latents = torch.randn(
        (1, pipe.unet.in_channels, height // 8, width // 8),
        generator = generator,
        dtype=torch.float16,
        device = device
    )
    latents = image_latents if latents is None else torch.cat((latents, image_latents))

PipelineOut = pipe(prompt, latents=latents, num_images_per_prompt=num_images_per_prompt)

### System Info

- `diffusers` version: 0.10.2
- Platform: Linux-5.4.0-122-generic-x86_64-with-glibc2.31
- Python version: 3.9.13
- PyTorch version (GPU?): 1.12.0+cu116 (True)
- Huggingface_hub version: 0.11.1
- Transformers version: 4.20.1
- Using GPU in script?: RTX A6000
- Using distributed or parallel set-up in script?: No multi GPUs or distributed

**Comment by chesha1** [comment]
It returns to normal automatically after I change the machine...
